### ISSUE TITLE
`vue/custom-event-name-casing`: should check and report `update:kebab-case`/`update:camelCase` event names too

### ISSUE THREAD

**Issue by zfeher** [vuejs/eslint-plugin-vue#2439]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.57.0 
- **eslint-plugin-vue version:** 9.24.0
- **Vue version:** 3.4.21
- **Node version:** 18.18.0
- **Operating System:** Windows 10

**Please show your full configuration:**

```json5
import pluginVue from 'eslint-plugin-vue';
import configPrettier from 'eslint-config-prettier';

export default [
  ...pluginVue.configs['flat/recommended'],
  configPrettier,

  {
    rules: {
      'vue/require-default-prop': 'off',
      'vue/attribute-hyphenation': ['error', 'never'],
      'vue/custom-event-name-casing': ['error', 'camelCase'],
      // 'vue/custom-event-name-casing': ['error', 'kebab-case'],
      'vue/v-on-event-hyphenation': ['error', 'never', { autofix: true }],
    },
  },
];
```

**What did you do?**	

```vue
<script setup>
const emit = defineEmits(['update:model-value',]);

const handleEmitClick = () => {
  emit('update:model-value', 'new foo');
};
</script>

<template>
  <button type="button" @click="$emit('update:model-value', 'new foo')">
    Emit
  </button>
</template>

```

**What did you expect to happen?**
`emit` and `$emit` calls with `update:kebab-case` like event names should be reported because of wrong event name casing but no error is reported. 
side note: `vue/v-on-event-hyphenation` are able to report such event names as error as expected.

**What actually happened?**

Nothing, no error is reported.
(It seems the rule only works with plain `kebab-case`/`camelCase` event names.)

**Repository to reproduce this issue**

[Repo with minimal reproduction](https://stackblitz.com/edit/vitejs-vite-w2xrze?file=eslint.config.js,src%2FApp.vue,src%2Fcomponents%2FFooBar.vue&view=editor)
Just open a new terminal or terminate dev server in current terminal then run `npm run lint` to see the problem. Feel free to change the eslint rule setting to `kebab-case` it won't report errors either.

**Comment by waynzh** [comment]
The [source code](https://github.com/vuejs/eslint-plugin-vue/blob/master/lib/rules/custom-event-name-casing.js#L121) and [test case](https://github.com/vuejs/eslint-plugin-vue/blob/master/tests/lib/rules/custom-event-name-casing.js#L72) look like it's intentional. 🤔

**Comment by ThiloCode** [comment]
@waynzh I just ran into this too, and would suggest it is a bug, for the following reason: Vue itself performs event name remapping even when the event name starts with `update:`.  So, if in a component I write `emit('update:myCustomEvent')`, I can create a listener for this event with either `@update:my-custom-event` or `@myCustomEvent`. I would argue that if Vue is doing this, the lint rule should be able to verify it too. 

Additionally I noticed that the Utils `casing.js` performs the following check

```function isKebabCase(str) {
  return (
    !hasUpper(str) &&
    !hasSymbols(str) &&
    !str.startsWith('-') && // starts with hyphen is not kebab-case
    !/_|--|\s/u.test(str)
  )
}
```

So anything containing symbols i.e. the `:` in `update:model-value` is registered as a symbol, which would cause the check to fail. I think the code just takes an unfortunate shortcut by ignoring every event starting with `update:` as seen here:

```
function isValidEventName(name) {
  return caseChecker(name) || name.startsWith('update:')
}
```

I'm new around here, but would be happy to try and contribute a fix if that's all right?

**Comment by FloEdelmann** [comment]
> I can create a listener for this event with […] or `@myCustomEvent`

I think you mean `@update:myCustomEvent` there, right?

**Comment by ThiloCode** [comment]
> > I can create a listener for this event with […] or `@myCustomEvent`
> 
> I think you mean `@update:myCustomEvent` there, right?

You're right, I've edited the initial comment to fix it.

**Comment by FloEdelmann** [comment]
Is it specific to `update:X` though? Or are all colon-separated camelCase event names working as described above?

**Comment by susnux** [comment]
Would you accept a PR fixing this?

IMHO this exception in the plugin logic is not needed, if you really want to allow `update:my-prop` instead of `update:myProp` then you can just use the `ignores` option this plugin provides  :)

**Comment by zfeher** [comment]
@susnux we would like to enforce either form for the whole code base as a convention. `ignores` option won't help with that unfortunately. having a convention helps searching/discovering these events without to need to try with both casing.

**Comment by susnux** [comment]
@zfeher that is what I meant.
So asking if the project owner of this plugin would accept a PR to fix this issue.
Because if you need `update:my-prop` you can just use that ignores option, but if you want to enforce `update:myProp` then this is currently not possible.

**Comment by waynzh** [comment]
@susnux PR welcome!
